FEXCore/Source/Interface/HLE/Thunks/Thunks.cpp

```diff
@@ -180,8 +180,25 @@ namespace FEXCore {
             Thread->CurrentFrame->State.gregs[FEXCore::X86State::REG_RDI] = (uintptr_t)arg0;
             Thread->CurrentFrame->State.gregs[FEXCore::X86State::REG_RSI] = (uintptr_t)arg1;
           } else {
+            // The argument pointer comes from the host stack, so it's not
+            // accessible by the guest. Allocate a thread-local page of memory
+            // to relocate the argument data.
+            // TODO: Directly store the arguments in a guest-accessible location instead
+            thread_local void* local_args = std::invoke([]() {
+              return mmap(
+                  0, 1000,
```

Comment:
comment says page but allocating only 1000 bytes through mmap instead of 0x1000.

Reply:
This remark got me reconsidering why I was using `mmap` here in the first place... Annoyingly enough, `FEXCore::Allocator::malloc` still returns something like `0xffffxxxxxxxx` here. mmap seemed to be the only reliable way to allocate guest-accessible memory.

I'll keep using mmap here, but changed over to the same amount copied below. (Also removed the unnecessary layer of immediately-invoked lambda.)